ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -276,11 +298,24 @@ public boolean updateAccounts(Collection<Account> accounts) {
   @Override
   public void close() {
     if (isOpen) {
-      try {
-        isOpen = false;
+      isOpen = false;
+      if (scheduler != null) {
+        scheduler.shutdown();
+        try {
+          if (!scheduler.awaitTermination(storeConfig.accountUpdaterShutDownTimeoutMs, TimeUnit.MILLISECONDS)) {
+            logger.warn("Scheduler for account updater is not shut down successfully");
+            scheduler.shutdownNow();
+            if (!scheduler.awaitTermination(storeConfig.accountUpdaterShutDownTimeoutMs, TimeUnit.MILLISECONDS)) {
+              logger.error("Scheduler for account updater is not shut down successfully");
+            }
+          }
+        } catch (InterruptedException e) {
+          scheduler.shutdownNow();
+          Thread.currentThread().interrupted();
+        } catch (Exception e) {
+          logger.error("Exception occurred when closing HelixAccountService.", e);
+        }
         helixStore.stop();
```

Comment:
should this be inside of the try catch like before?

Reply:
Ok, it might be better to put it back, though it does not explicitly throw any exception.

Reply:
Need to get it out as now we have modularized code for shutting down executorservice.